plugins/rest/rest_handlers.go

```diff
@@ -29,175 +29,151 @@ import (
-	l2plugin "github.com/ligato/vpp-agent/plugins/vpp/l2plugin/vppcalls"
 	l3plugin "github.com/ligato/vpp-agent/plugins/vpp/l3plugin/vppcalls"
 	"github.com/ligato/vpp-agent/plugins/vpp/model/acl"
+	"github.com/ligato/vpp-agent/plugins/vpp/model/interfaces"
 )
 
-// interfacesGetHandler - used to get list of all interfaces
-func (plugin *Plugin) interfacesGetHandler(formatter *render.Render) http.HandlerFunc {
-	return func(w http.ResponseWriter, req *http.Request) {
-
-		plugin.Log.Debug("Getting list of all interfaces")
-
-		// create an API channel
-		ch, err := plugin.GoVppmux.NewAPIChannel()
-		if err != nil {
-			plugin.Log.Errorf("Error creating channel: %v", err)
-			formatter.JSON(w, http.StatusInternalServerError, err)
-			return
-		}
-		defer ch.Close()
-
-		ifHandler, err := ifplugin.NewIfVppHandler(ch, plugin.Log, nil)
-		if err != nil {
-			plugin.Log.Errorf("Error creating VPP handler: %v", err)
-			formatter.JSON(w, http.StatusInternalServerError, err)
-			return
-		}
-		res, err := ifHandler.DumpInterfaces()
-		if err != nil {
-			plugin.Log.Errorf("Error: %v", err)
-			formatter.JSON(w, http.StatusInternalServerError, err)
-			return
-		}
-
-		plugin.Log.Debug(res)
-		formatter.JSON(w, http.StatusOK, res)
-	}
+// Registers access list REST handlers
+func (plugin *Plugin) registerAccessListHandlers() {
+	// GET IP ACLs
+	plugin.registerHTTPHandler(resturl.AclIP, GET, func() (interface{}, error) {
+		return plugin.aclHandler.DumpIPACL(nil)
+	})
+	// GET MACIP ACLs
+	plugin.registerHTTPHandler(resturl.AclMACIP, GET, func() (interface{}, error) {
+		return plugin.aclHandler.DumpMACIPACL(nil)
+	})
+	// GET IP ACL example
+	plugin.HTTPHandlers.RegisterHTTPHandler(resturl.AclIPExample, plugin.exampleIpACLGetHandler, GET)
+	// GET MACIP ACL example
+	plugin.HTTPHandlers.RegisterHTTPHandler(resturl.AclMACIPExample, plugin.exampleMacIpACLGetHandler, GET)
 }
 
-// bridgeDomainIdsGetHandler - used to get list of all bridge domain ids
-func (plugin *Plugin) bridgeDomainIdsGetHandler(formatter *render.Render) http.HandlerFunc {
-	return func(w http.ResponseWriter, req *http.Request) {
-
-		plugin.Log.Debug("Getting list of all bridge domain ids")
-
-		// create an API channel
-		ch, err := plugin.GoVppmux.NewAPIChannel()
-		if err != nil {
-			plugin.Log.Errorf("Error creating channel: %v", err)
-			formatter.JSON(w, http.StatusInternalServerError, err)
-			return
+// Registers interface REST handlers
+func (plugin *Plugin) registerInterfaceHandlers() {
+	// GET all interfaces
+	plugin.registerHTTPHandler(resturl.Interface, GET, func() (interface{}, error) {
+		return plugin.ifHandler.DumpInterfaces()
+	})
+	// GET loopback interfaces
+	plugin.registerHTTPHandler(resturl.Loopback, GET, func() (interface{}, error) {
+		ifs, err := plugin.ifHandler.DumpInterfaces()
+		for ifKey, ifConfig := range ifs {
+			if ifConfig.Interface.Type != interfaces.InterfaceType_SOFTWARE_LOOPBACK {
```

Comment:
This filtering of interfaces seems to be duplicated enough times for it to be extracted as separate function. Something like: `plugin.DumpInterfacesType(interfaces.InterfaceType_SOFTWARE_LOOPBACK)`